### PR TITLE
Improve __eq__ functions in raid classes

### PR DESCRIPTION
**Pull request by Kuchenmampfer**
The old `RaidClan.__eq__` only returned true if the attacks on it where identical. Because the `__eq__` functions of `RaidDistrict` and `RaidAttack` checked if their parent `RaidClan` was identical, **all** the attacks and districts said they were different as soon as one additional attack on any district was made. I say this is a bug because only that district/attack changed and the others should have their `__eq__` return true when compared with the same attack/district from before. This is what this PR fixes. Additionally, I improved the code style and removed some redundant checks.

### CODE REVIEW

**Comment by Kuchenmampfer** [comment]
`RaidMember` is the only thing I left untouched so far, but the code style can also be improved there and I think the `__eq__` should only compare the members tag and the parent RaidLogEntry because if the member does an attack, its still the same member. Has anyone a different opinion or can I also change this?

**Comment by MagicTheDev** [comment]
I agree, attacks shouldn't be part of the comparison

**Comment by Kuchenmampfer** [comment]
I looked into the `__eq__` of `RaidLogEntry` as well and think it should only check for clan tag and start time. So I will merge magics clan tag addition into this I think.

File: coc/raid.py [hunk 1]
```diff
@@ -161,15 +155,11 @@ def __repr__(self):
         return "<%s %s>" % (self.__class__.__name__, " ".join("%s=%r" % t for t in attrs),)
 
     def __eq__(self, other):
-        if isinstance(other, RaidAttack):
-            if (self.attacker_tag == other.attacker_tag
+        return (isinstance(other, RaidAttack)
+                and self.attacker_tag == other.attacker_tag
                 and self.destruction == other.destruction
-                and self.stars == other.stars
```
Comment (lukasthaler) [inline coc/raid.py:167]: Why are we removing the stars from the comparison?

Comment (doluk) [reply coc/raid.py:167]: because magic and ü think it should be removed. I disagree and because I wrote the code initially it is there and to be removed

Comment (lukasthaler) [reply coc/raid.py:167]: I wouldn't remove the stars here. It costs us nothing to keep them in (the check is extremely inexpensive) and shuts the door on any potential false positives

Comment (doluk) [reply coc/raid.py:167]: I mean it should be not possible to get the same destruction with more stars, but you never know🤣

Comment (Kuchenmampfer) [reply coc/raid.py:167]: I think its not needed, but it also does not matter. If you want to keep it we can keep it

**Pull request merged**